onboarding/source/amb_light_service.c

```diff
@@ -1,39 +1,134 @@
 #include "amb_light_service.h"
 #include "serial_io.h"
+#include <stdio.h>
 
 #include <adc.h>
 #include <sci.h>
+#include <FreeRTOS.h>
+#include <os_task.h>
+
+#include <os_queue.h>
+#include <os_projdefs.h>
+#include <string.h>
+
+#define LIGHT_SENSOR 	6U
 
 /* USER CODE BEGIN */
 // Include any additional headers and global variables here
 
+static TaskHandle_t lightServiceHandle = NULL;
+static QueueHandle_t event_queue = NULL;
 /* USER CODE END */
 
 /**
  * @brief Task that waits for MEASURE_LIGHT event and then prints the ambient light value to the serial port.
  * @param pvParameters Task parameters
  */
+
+void adcStartConversion_selChn(adcBASE_t *adc, uint8_t channel, uint8_t fifo_size, uint8_t group);
+void adcGetSingleData(adcBASE_t *adc, uint8_t group, adcData_t *data);
+uint32_t getLightSensorData(void);
 static void lightServiceTask(void * pvParameters);
 
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    
+    if (event_queue == NULL) {
+        event_queue = xQueueCreate(1, LIGHT_SERVICE_QUEUE_SIZE);
+    }
 
-    /* USER CODE END */
+    if (lightServiceHandle == NULL) {
+        // Create controller task
+        xReturned = xTaskCreate(lightServiceTask,             /* Function that implements the task. */
+                                CONTROLLER_NAME_LIGHT,            /* Text name for the task. */
+                                CONTROLLER_STACK_SIZE,      /* Stack size in words, not bytes. */
+                                NULL,                       /* Parameter passed into the task. */
+                                CONTROLLER_PRIORITY,        /* Priority at which the task is created. */
+                                &lightServiceHandle);     /* Used to pass out the created task's handle. */
+    }
+
+    if(xReturned == pdFAIL){
+        unsigned char err[] = "Error Starting Light Service";
+        sciPrintText(scilinREG, (unsigned char*) err, strlen((const char*) err));
+    }
+
+    if(event_queue == NULL){
+        unsigned char err[] = "Error Starting Queue";
+        sciPrintText(scilinREG, (unsigned char*) err, strlen((const char*) err));
+    }    
+    /* USER CODE END */ 
     return 1;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceHandle != NULL); 
+    ASSERT(event_queue != NULL);
+    light_event_t eventBuffer;
+
+    uint8_t lightServiceStatus = initLightService();
+
+    if(!lightServiceStatus) {
+        unsigned char err[] = "Error Starting Light Service";
+        sciPrintText(scilinREG, (unsigned char*) err, strlen((const char*) err));
+    }
+
+    while(true) {
+        if(xQueueReceive(event_queue, &eventBuffer, ( TickType_t ) 0) == pdPASS) {
```

Comment:
You should use a non-zero block time for xQueueReceive. If it's 0, then we'll basically be polling the queue which would result in task starvation if we had lower-priority tasks.